contracts/icp/context-config/src/mutate.rs

```diff
@@ -83,12 +89,53 @@ fn add_context(
             return Err("context already exists".into());
         }
 
-        configs.next_proxy_id += 1;
-
         Ok(())
     })
 }
 
+async fn deploy_proxy_contract(context_id: &ICContextId) -> Result<Principal, String> {
+    // Get the proxy code
+    let proxy_code = CONTEXT_CONFIGS
+        .with(|configs| configs.borrow().proxy_code.clone())
+        .ok_or("proxy code not set")?;
+
+    // Create canister with cycles
+    let create_args = CreateCanisterArgument {
+        settings: Some(CanisterSettings {
+            controllers: Some(vec![ic_cdk::api::id()]),
+            compute_allocation: None,
+            memory_allocation: None,
+            freezing_threshold: None,
+            reserved_cycles_limit: None,
+            log_visibility: None,
+            wasm_memory_limit: None,
+        }),
+    };
+
+    let (canister_record,) = create_canister(create_args, 500_000_000_000_000u128)
```

Comment:
should this cycle count be dynamic?

Reply:
this is minimum what we need to transfer to the created contract for it to work.